src/App.jsx

```diff
@@ -1,60 +1,142 @@
+import { useState } from "react"
 import "./App.css"
 
 function App() {
+  const [numberLeft, setNumberLeft] = useState(0);
+  const [numberRight, setNumberRight] = useState(0);
+  const [result, setResult] = useState(0);
+
+  const [operator, setOperator] = useState("");
+
+  const [storeResult, setStoreResult] = useState(0);
+  const [storeLeft, setStoreLeft] = useState(0);
+  const [storeRight, setStoreRight] = useState(0);
+
+
+  const handleNumberClickLeft = (e) => {
+    const clickedNumber = e.target.innerText;
+    if (clickedNumber === "." && numberLeft.toString().includes(".")) {
+      return;
+    }
+    setNumberLeft((prevNumber) => parseFloat(prevNumber + clickedNumber));
+  };
+
+  const handleNumberClickRight = (e) => {
+    const clickedNumber = e.target.innerText;
+    if (clickedNumber === "." && numberRight.toString().includes(".")) {
+      return;
+    }
+    setNumberRight((prevNumber) => parseFloat(prevNumber + clickedNumber));
+  };
+
+  const handleDecimalClickLeft = () => {
+    if (!numberLeft.toString().includes(".")) {
+      setNumberLeft((prevNumber) => prevNumber + ".");
+    }
+  };
+
+  const handleDecimalClickRight = () => {
+    if (!numberRight.toString().includes(".")) {
+      setNumberRight((prevNumber) => prevNumber + ".");
+    }
+  };
+
+  const handleStoreClick = () => {
+    setResult(storeResult);
+  }
+
+  const handleStoreRightClick = () => {
+    setNumberRight(storeRight);
+  }
+
+  const handleStoreLeftClick = () => {
+    setNumberLeft(storeLeft);
+  }
+
+  function resetNumbers() {
+    setNumberLeft(0);
+    setNumberRight(0);
+  }
+
+
+  const handleClick = (e) => {
+    const operator = e.target.innerText;
+    if (operator === "+" || operator === "-" || operator === "*" || operator === "÷") {
+      setOperator(operator);
+    }
+    setStoreLeft(numberLeft);
+    setStoreRight(numberRight);
+    if (operator === "=") {
+      setResult(numberLeft + numberRight);
+    } else if (operator === "Clear") {
+      setResult(0);
+    } else if (operator === "+") {
+      setResult(numberLeft + numberRight);
+      setStoreResult(numberLeft + numberRight)
+    } else if (operator === "-") {
+      setResult(numberLeft - numberRight);
+      setStoreResult(numberLeft - numberRight)
+    } else if (operator === "*") {
+      setResult(numberLeft * numberRight);
+      setStoreResult(numberLeft * numberRight)
+    } else if (operator === "÷") {
+      setResult(numberLeft / numberRight);
+      setStoreResult(numberLeft / numberRight)
+    } else {
+      setResult(0);
+      setStoreResult(0)
+    }
+    resetNumbers();
```

Comment:
why are you resetting when you compute result? might be nice to keep previous numbers